src/lib/SCEditor.js

```diff
@@ -379,6 +380,35 @@ export default function SCEditor(original, userOptions) {
 	// Don't deep extend emoticons (fixes #565)
 	base.opts.emoticons = userOptions.emoticons || defaultOptions.emoticons;
 
+	// Allow YouTube iframes
+	// https://github.com/cure53/DOMPurify/issues/340#issuecomment-670758980
+	DOMPurify.addHook('uponSanitizeElement', (node, data) => {
+		if (data.tagName === 'iframe') {
+			var src = node.getAttribute('src') || '';
+			if (src.substr(0, 39) !== 'https://www.youtube-nocookie.com/embed/') {
```

Comment:
Will need some way to specify allowed iframe URLs.

This will it will require a major version dump as it won't be fully backwards compatible if using any plugins that embed iframes as they will need to be updated to specify allowed URLs.